### ISSUE TITLE
Enable console output in productive build. 

### ISSUE THREAD

**Issue by snmed** [wailsapp/wails#544]
**Description**
I'm building an app that can be used with a gui and also as a cli tool. Unfortunately, I can't get any 
console output, except if I build it with `-d` as debug binary. But I don't want debug output when I
run in gui mode. Is there any way to support console output in productive builds?

Is it possible or are there any system limitations that prevents the use of console output in a productive build.

Any help would be greatly appreciated.

**To Reproduce**
main.go:
```
	// Check if we are called with command line arguments
	if len(os.Args) >= 2 {
		if os.Args[1] == "run" && len(os.Args) >= 3 {

			if err := cerberus.RunService(os.Args[2]); err != nil {
				log.Fatalln(err)
			}
			os.Exit(0)
		}

	} else {

		js := mewn.String("./frontend/dist/app.js")
		css := mewn.String("./frontend/dist/app.css")

		svcs := &Services{}

		app := wails.CreateApp(&wails.AppConfig{
			Width:     1024,
			Height:    768,
			Title:     "cerberus-sm",
			JS:        js,
			CSS:       css,
			Colour:    "#131313",
			Resizable: true,
		})

		app.Bind(svcs)
		app.Run()
	}
```

**Expected behaviour**
Output in console if  `fmt.Println`  is used.

**System Details**
- Windows 10
- go 1.15.2
- TDM-GCC 1.2001.0

- [x ] This issue is for Windows and I have read the [Windows Developer Guide](https://wails.app/guides/windows/)

**Comment by leaanthony** [comment]
By default, Windows prod builds use the `-H windowsgui` flag. We can provide an override for that :+1:

**Comment by snmed** [comment]
That sounds promising, but arises some questions:

a. If we build it without `-H windowsgui`, am I right that as soon as I double click the executable, a console windows pops up?
b. I surmise the override must be done at compile time and not at runtime?
c. If a and b are true, isn't there another option for example attaching some `stdin`  and `stdout` to the process? Because in gui mode it isn't appealing to have an open console window.

If I can help in any way let me know.

**Comment by leaanthony** [comment]
In answer to your questions:

a. Yes, I believe so
b. Yes, compile time
c. I'm struggling to understand what your need is. Where would the console be if not in a window?

**Comment by snmed** [comment]
For c. probably my question wasn't that clear, because you probably already recognized, English is not my native language.

I'll try to elaborate my question, if I start the application with a double click, I just want the GUI window and not an additional console window. On the other hand, if i start the application with a command line argument ex. `myapp.exe cli [more arguments]` in a console, printing to console should be possible.

I read through some articles and eventually that's not possible. Or do you see a way to achieve that particular behaviour?

**Comment by leaanthony** [comment]
On windows I'm not sure that is possible. Consider logging the information to a file?

**Comment by snmed** [comment]
It's not just about logging, it's about print the cli options and parameters to use instead the gui itself. Anyway, in my case I will omit the cli tools for the moment, because I need primarily the `run` command to execute the windows service handler in the background.

Thank you very much for your help and quick responses.

**Comment by campbellwmorgan** [comment]
@leaanthony I know this issue was closed a while ago, but it would be incredibly useful for me to have an option where I can disable the setting of the `wingui` `ldflags` in production. I would happy to submit a PR for this if you think you think it would be appropriate?

In the meantime, I've found a workaround for this which is to use `editbin.exe` with the `/SUBSYSTEM:CONSOLE` flag to force a console (https://docs.microsoft.com/en-us/cpp/build/reference/subsystem?view=msvc-170)

**Comment by leaanthony** [comment]
Happy to talk about the use case for this!

**Comment by campbellwmorgan** [comment]
Similar to the OP above I need both the cli stdout and the GUI to work. My problem is that the project I'm working on is cli first for programmatic consumption, but then has a gui which is used by some customers. Two separate executables is unwieldy hence the desire to combine both together. The debug flag does everything I want except that it allows users to inspect Dom elements which in an ideal world I would like to restrict. As I mentioned above I have a good workaround using a production build and `editbin.exe` in my CI but it would be nice to not have to do that and just use a flag in my wails config.

Did not mention by the way, but it's a great project! Thanks so much for your work.

**Comment by leaanthony** [comment]
So let me get this straight, you'd all like a flag to leave the console visible in prod mode?

**Comment by campbellwmorgan** [comment]
@leaanthony Yes exactly that! As I said, if you wouldn't mind that, I would be happy to submit a PR to that effect

**Comment by leaanthony** [comment]
I've pushed support for this on [master](https://wails.io/docs/next/guides/bleeding-edge).
The docs will be updated [here ](https://wails.io/docs/next/reference/cli/) shortly.
tl;dr; 
  - Once you've got a [local development environment](https://wails.io/docs/next/contributing/setting_up_a_dev_environment), build your app with `wails build -windowsconsole`
